File: frontend-v1/src/models/JwtResponse.js
/**
 * Data Transfer Object for JWT authentication response
 * This model matches the backend JwtResponse.java class
 */
class JwtResponse {
  constructor({
    token,
    type = 'Bearer',
    id,
    username,
    email,
    role,
    refreshToken
  }) {
    this.token = token;
    this.type = type;
    this.id = id;
    this.username = username;
    this.email = email;
    this.role = role;
    this.accessToken = token; // Alias for token for compatibility
    this.refreshToken = refreshToken;
  }

  /**
   * Get the full authorization header value
   * @returns {string} Authorization header value
   */
  getAuthorizationHeader() {
    return `${this.type} ${this.token}`;
  }

  /**
   * Create a JwtResponse instance from API response
   * @param {Object} response - API response data
   * @returns {JwtResponse} New JwtResponse instance
   */
  static fromResponse(response) {
    return new JwtResponse({
      token: response.token || response.accessToken,
      type: response.type || 'Bearer',
      id: response.id,
      username: response.username,
      email: response.email,
      role: response.role,
      refreshToken: response.refreshToken
    });
  }
}

export default JwtResponse;

File: services/user_service/src/main/java/com/user/user_service/dto/JwtResponse.java
package com.user.user_service.dto;

import com.user.user_service.model.Role;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

/**
 * DTO for JWT authentication response
 */
@Data
@Builder
@AllArgsConstructor
@NoArgsConstructor
public class JwtResponse {
    private String token;
    private String type = "Bearer";
    private Long id;
    private String username;
    private String email;
    private Role role;
    
    public JwtResponse(String token, Long id, String username, String email, Role role) {
        this.token = token;
        this.id = id;
        this.username = username;
        this.email = email;
        this.role = role;
    }
}